Building local docker containers with make docker-build is broken

I am running into an error with `make docker-build` because the binary is not found:

```
$ make docker-build
docker build -t boot-service:latest .
STEP 1/8: FROM gcr.io/distroless/static-debian12:nonroot
Trying to pull gcr.io/distroless/static-debian12:nonroot...
Getting image source signatures
Copying blob 7c12895b777b done   | 
Copying blob 526604835308 done   | 
Copying blob ef49c20a7b35 done   | 
Copying blob 990a9c434e5e done   | 
Copying blob 2780920e5dbf done   | 
Copying blob bf7a4185f015 done   | 
Copying blob 3214acf345c0 done   | 
Copying blob 52630fc75a18 done   | 
Copying blob dd64bf2dd177 done   | 
Copying blob b839dfae01f6 done   | 
Copying blob dcaa5a89b0cc done   | 
Copying blob 069d1e267530 done   | 
Copying config 3a59ed7ba5 done   | 
Writing manifest to image destination
STEP 2/8: WORKDIR /app
--> 26cfc71b3d2c
STEP 3/8: COPY boot-service /usr/local/bin/boot-service
Error: building at STEP "COPY boot-service /usr/local/bin/boot-service": checking on sources under "/opt/shared/boot-service": copier: stat: "/boot-service": no such file or directory
make: *** [Makefile:58: docker-build] Error 125
```

A couple of suggestions:

1. Generalize container build target (e.g. `container-build`/`container-run`), allowing other runtimes other than Docker (e.g. Podman) for builds, as well as specifying tags, etc. (See [[1]](https://github.com/OpenCHAMI/coresmd/blob/7eb33cebc4c4872f129d0c0094f0285ff1d0986b/Makefile#L17) [[2]](https://github.com/OpenCHAMI/coresmd/blob/7eb33cebc4c4872f129d0c0094f0285ff1d0986b/Makefile#L43) [[3]](https://github.com/OpenCHAMI/coresmd/blob/7eb33cebc4c4872f129d0c0094f0285ff1d0986b/Makefile#L45) [[4]](https://github.com/OpenCHAMI/coresmd/blob/7eb33cebc4c4872f129d0c0094f0285ff1d0986b/Makefile#L87-L92))
2. Either combine the Dockerfiles or have the container build target use the standalone one.
   - If we want to separate the Goreleaser Dockerfile from the multi-stage build Dockerfile, we should have the container build Make target use the latter. Else if we consolidate the Dockerfiles, we should have the build Make target put the binaries in the repo root and name them (as Goreleaser does, e.g. `boot-service` and `boot-service-client`) so the single Dockerfile will find them (we'll also need to make sure the container build target lists the binaries as prerequisites)
   - CoreSMD uses a consolidated Dockerfile. See [[4]](https://github.com/OpenCHAMI/coresmd/blob/7eb33cebc4c4872f129d0c0094f0285ff1d0986b/Makefile#L87-L92) [[5]](https://github.com/OpenCHAMI/coresmd/blob/7eb33cebc4c4872f129d0c0094f0285ff1d0986b/Makefile#L204-L214)

_Originally posted by @synackd in https://github.com/OpenCHAMI/boot-service/issues/7#issuecomment-4181026247_

--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 #
 # SPDX-License-Identifier: MIT
 
-.PHONY: help build test test-integration lint clean install run docker-build docker-run release-test check-no-pkg-resources-imports generate generate-check dev
+.PHONY: help build test test-integration lint clean install run container-build container-run release-test check-no-pkg-resources-imports generate generate-check dev
 
 # Variables
 BINARY_NAME=boot-service
@@ -13,7 +13,8 @@ TEST_TIMEOUT ?= 5m
 VERSION ?= $(shell git describe --tags --always --dirty 2>/dev/null || echo "dev")
 COMMIT ?= $(shell git rev-parse --short HEAD 2>/dev/null || echo "unknown")
 DATE ?= $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
-DOCKER_GO_VERSION ?= $(shell awk '/^go / {print $$2; exit}' go.mod)
+CONTAINER_PROG ?= $(shell command -v docker 2>/dev/null)
+CONTAINER_GO_VERSION ?= $(shell awk '/^go / {print $$2; exit}' go.mod)
 FABRICA_VERSION ?= $(shell awk '/github.com\/openchami\/fabrica[[:space:]]+v/ {print $$2; exit}' go.mod)
 LDFLAGS=-ldflags "-X main.version=$(VERSION) -X main.commit=$(COMMIT) -X main.date=$(DATE)"
 FABRICA_CMD ?= go run github.com/openchami/fabrica/cmd/fabrica@$(FABRICA_VERSION)
@@ -88,16 +89,16 @@ tidy: ## Tidy go.mod
 run: build ## Build and run the application
 	./bin/$(BINARY_NAME)
 
-docker-build: ## Build Docker image
-	docker build -f Dockerfile.standalone \
-		--build-arg GO_VERSION=$(DOCKER_GO_VERSION) \
+container-build: ## Build container image
+	$(CONTAINER_PROG) build -f Dockerfile.standalone \
+		--build-arg GO_VERSION=$(CONTAINER_GO_VERSION) \
 		--build-arg VERSION=$(VERSION) \
 		--build-arg COMMIT=$(COMMIT) \
 		--build-arg DATE=$(DATE) \
 		-t $(BINARY_NAME):latest .
 
-docker-run: docker-build ## Build and run Docker container
-	docker run --rm $(BINARY_NAME):latest
+container-run: container-build ## Build and run container
+	$(CONTAINER_PROG) run --rm $(BINARY_NAME):latest
 
 release-snapshot: ## Create a snapshot release with GoReleaser
 	goreleaser release --snapshot --clean

